pymc_marketing/product_incrementality/mv_its.py

```diff
@@ -0,0 +1,573 @@
+        x = self.X.index.values
+        background_products = self.coords["background_product"]
+
+        for i, background_product in enumerate(background_products):
+            az.plot_hdi(
+                x,
+                self.causal_impact(variable=variable)
+                .transpose(..., "time")
+                .sel(background_product=background_product),
+                fill_kwargs={
+                    "alpha": HDI_ALPHA,
+                    "color": f"C{i}",
+                    "label": "Posterior predictive (HDI)",
+                },
+                smooth=False,
+            )
+        ax.set(ylabel="Change in sales caused by new product")
+
+        # formatting
+        ax.legend()
+        ax.set(title="Estimated causal impact of new product upon existing products")
+        return ax
+
+    def plot_causal_impact_market_share(self, variable="mu"):
+        """Plot the inferred causal impact of the new product on the background products.
+
+        Note: if we compare "mu" then we are comparing the expected sales, if we compare
+        "y" then we are comparing the actual sales
+        """
+        _, ax = plt.subplots()
+
+        # plot posterior predictive distribution of sales for each of the background products
+        x = self.X.index.values
+        background_products = list(self.idata.observed_data.background_product.data)
+
+        # divide the causal impact change in sales by the counterfactual predicted sales
+        variable = "mu"
+        for i, background_product in enumerate(background_products):
+            causal_impact = (
+                self.causal_impact(variable=variable)
+                .transpose(..., "time")
+                .sel(background_product=background_product)
+            )
+            total_sales = (
+                self.idata.predictions[variable]
+                .transpose(..., "time")
+                .sum(dim="background_product")
+            )
+            causal_impact_market_share = (causal_impact / total_sales) * 100
+
+            az.plot_hdi(
+                x,
+                causal_impact_market_share,
+                fill_kwargs={
+                    "alpha": HDI_ALPHA,
+                    "color": f"C{i}",
+                    "label": f"{background_product} - Posterior predictive (HDI)",
+                },
+                smooth=False,
+            )
+        ax.set(ylabel="Change in market share caused by new product")
+        ax.yaxis.set_major_formatter(mtick.PercentFormatter())
+
+        # formatting
+        ax.legend()
+        ax.set(title="Estimated causal impact of new product upon existing products")
+        return ax
+
+    def plot_data(self, ax=None):
+        """Plot the observed data."""
+        data = pd.concat([self.X, self.y], axis=1)
+
+        return plot_product(data=data, ax=ax)
+
+
+def plot_product(data: pd.DataFrame, ax: plt.Axes | None = None) -> plt.Axes:
```

Comment:
Total sales with many products is on a different scale. I think this should be a flag and expose it in the various plotting methods

Reply:
Agree!

Reply:
resolved by https://github.com/pymc-labs/pymc-marketing/pull/803/commits/704fc2577dff6e4c9e7d7637a62eb7123be5ec52